ui/pages/confirmations/components/confirm/info/approve/edit-spending-cap-modal/edit-spending-cap-modal.tsx

```diff
@@ -0,0 +1,138 @@
+}: {
+  isOpenEditSpendingCapModal: boolean;
+  setIsOpenEditSpendingCapModal: (newValue: boolean) => void;
+  customSpendingCap: string;
+  setCustomSpendingCap: (newValue: string) => void;
+}) => {
+  const t = useI18nContext();
+
+  const { currentConfirmation: transactionMeta } = useConfirmContext() as {
+    currentConfirmation: TransactionMeta;
+  };
+
+  const { userBalance, tokenSymbol, decimals } = useAssetDetails(
+    transactionMeta.txParams.to,
+    transactionMeta.txParams.from,
+    transactionMeta.txParams.data,
+  );
+
+  const accountBalance = getAccountBalance(userBalance || '0', decimals || '0');
+
+  const { formattedSpendingCap } = useApproveTokenSimulation(
+    transactionMeta,
+    decimals || '0',
+  );
+
+  const [customSpendingCapCandidate, setCustomSpendingCapCandidate] =
+    useState('');
+
+  const onCancelHandler = () => {
+    setIsOpenEditSpendingCapModal(false);
+    setCustomSpendingCapCandidate('');
+  };
+
+  function submitEditSpendingCap() {
+    setIsOpenEditSpendingCapModal(false);
+    setCustomSpendingCapCandidate('');
+    setCustomSpendingCap(customSpendingCapCandidate);
+  }
+
+  return (
+    <Modal
+      isOpen={isOpenEditSpendingCapModal}
+      onClose={() => setIsOpenEditSpendingCapModal(false)}
+      isClosedOnEscapeKey
+      isClosedOnOutsideClick
+      className="edit-spending-cap-modal"
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader
+          justifyContent={JustifyContent.center}
+          childrenWrapperProps={{
+            alignItems: AlignItems.center,
+            display: Display.Flex,
+            flexDirection: FlexDirection.Column,
+          }}
+        >
+          <Text variant={TextVariant.headingMd}>{t('editSpendingCap')}</Text>
+        </ModalHeader>
+        <ModalBody>
+          <Text
+            variant={TextVariant.bodyMd}
+            color={TextColor.textAlternative}
+            paddingBottom={4}
+          >
+            {t('editSpendingCapDesc')}
+          </Text>
+          <TextField
+            type={TextFieldType.Number}
+            value={customSpendingCapCandidate}
+            onChange={(event) =>
+              setCustomSpendingCapCandidate(event.target.value)
+            }
+            placeholder={`${
+              customSpendingCap === ''
+                ? formattedSpendingCap
+                : customSpendingCap
+            } ${tokenSymbol}`}
+            style={{ width: '100%' }}
+            inputProps={{ 'data-testid': 'custom-spending-cap-input' }}
+          />
+          <Text
+            variant={TextVariant.bodySm}
+            color={TextColor.textAlternative}
+            paddingTop={1}
+          >
+            {t('editSpendingCapAccountBalance', [
+              accountBalance,
+              tokenSymbol || '',
+            ])}
+          </Text>
+        </ModalBody>
+        <ModalFooter
+          onSubmit={() => submitEditSpendingCap()}
+          onCancel={() => onCancelHandler()}
```

Comment:
Minor, there's no custom arguments needed here and they're already assigned to variables so no need for `() =>` here.